### ISSUE TITLE
Folder rename 

### ISSUE THREAD

**Issue by Fohio57** [vitiko98/qobuz-dl#227]
Hello,

I must be dumb but i can't change the downloaded album name

-ff PATTERN, --folder-format PATTERN
                        pattern for formatting folder names, e.g "{artist} -
                        {album} ({year})". available keys: artist,
                        albumartist, album, year, sampling_rate, bit_depth,
                        tracktitle, version. cannot contain characters used by
                        the system, which includes /:<>

How must i write correctly ?

qobuz-dl dl "url" -ff PATTERN "({year}) {artist} - {album} "

I just want my download will put the year in first.

thanks

**Comment by Qamynn** [comment]
I don't know if I now misunderstand you, but you don't need to actually put the word PATTERN in the command. It's the name of the variable. Your pattern is: "({year}) {artist} - {album}".

That means your command will be:
`qobuz-dl dl "url" -ff "({year}) {artist} - {album}"`

**Comment by Fohio57** [comment]
This is working perfectly. :)
Can this setting be changed permanently?

**Comment by Qamynn** [comment]
Yes you can. Not from just a command, you will have to change 1 line in one of the qobuz_dl files, but it's pretty simple.

1. First you look up where qobuz_dl is stored, you do that by running this command in command prompt/terminal:
`python -c "import qobuz_dl as _; print(_.__path__[0])"`

2. It will return a path and you should open that in file explorer and you should see a file called:
> downloader.py

3. (This is for windows) Right-click the file and click "Edit with IDLE" (or your preferred code editor). In IDLE, go to the options at the top and enable "Show Line Numbers".

4. Go to _**line 26**_ and it should be:
```py
DEFAULT_FOLDER = "{artist} - {album} ({year}) [{bit_depth}B-{sampling_rate}kHz]"
```

5. Then you can change `{artist} - {album} ({year}) [{bit_depth}B-{sampling_rate}kHz]` to any pattern you want.

This changes the default value permanently until the next update of qobuz_dl. You can still use `-ff` if you want to change the pattern for a single download.

**Comment by 0f4reeder** [comment]
OK made changes to line 26 but no joy...am using the following:
cd C:\Users\alree\Desktop

qobuz-dl dl this_txt_file_has_urls.txt --albums-only -q 27

**Comment by Qamynn** [comment]
Hm, I'm not sure why this is, I don't really remember why or how this works again, but I might have the solution. You might want to follow the steps above but open the file:
> core.py

And instead of going to line 26, go to **line 51/52** and it should be:
```py
folder_format="{artist} - {album} ({year}) [{bit_depth}B-"
        "{sampling_rate}kHz]",
```
When changing this, you don't need to divide the pattern in to 2 different string with an enter like they did here. Just change the pattern to whatever you want in quotation marks of course. Also don't forget to place the comma at the end. Like this for example:
```py
folder_format="({year}) {artist} - {album}",
```
Changing this could probably be this solution, sorry, but I haven't tested this.

**Comment by 0f4reeder** [comment]
Thanks but still no joy. My changes are:

downloader.py lines 26 & 27
DEFAULT_FOLDER = "{artist} - {album} {bit_depth}-{sampling_rate} [{year}]"
DEFAULT_TRACK = "{tracknumber} {tracktitle}"

and

core.py lines 53 & 53
        folder_format="{artist} - {album} {bit_depth}-{sampling_rate} [{year}]",
        track_format="{tracknumber} {tracktitle}",

question is this behavior somehow connected to the bulk download 

qobuz-dl dl this_txt_file_has_urls.txt --albums-only -q 27

These also not changed from defalt:

qobuz-dl dl https://play.qobuz.com/album/eh5prmm06py0b -q 27
qobuz-dl dl https://play.qobuz.com/artist/60885 -q 27 --no-db

**Comment by 0f4reeder** [comment]
Just in case it is some help the files changed are included and
an additional directory in site-packages

[Desktop.zip](https://github.com/vitiko98/qobuz-dl/files/12897010/Desktop.zip)

**Comment by 0f4reeder** [comment]
Perhaps I missed something. Attached are the .py files I have edited.

[Desktop.zip](https://github.com/vitiko98/qobuz-dl/files/12897247/Desktop.zip)

**Comment by 0f4reeder** [comment]
Since I load my downloads into text files I just add a -ff to the download.
Thankyou very much to all.
